builtin/providers/vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -421,9 +421,16 @@ func resourceVSphereVirtualMachine() *schema.Resource {
 							Default:  "scsi",
 							ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
 								value := v.(string)
-								if value != "scsi" && value != "ide" {
```

Comment:
I like not having to iterate! To do this wouldn't I have to redeclare the set of controller types inside of the validateFunc?  My mentality was to have them defined just once, so if a new controller type is supported later, it would only have to be added in one place.  

Is this about what you were thinking:
```
ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
	value := v.(string)
	validTypes := map[string]struct{}{
		"scsi": struct{}{},
		"scsi-lsi-parallel": struct{}{},
		"scsi-buslogic": struct{}{},
		"scsi-paravirtual": struct{}{},
		"scsi-lsi-sas": struct{}{},
		"ide": struct{}{},
	}
	if _, ok := validTypes[value]; !ok {
		errors = append(errors, fmt.Errorf("Supported values for 'controller_type are %v", strings.Join(DiskControllerTypes, ", ")))
	}
```

Do you find it better to redeclare the list to prevent an iteration here?  I'm on the fence so I'll let you decide.